coderd/workspaceagents.go

```diff
@@ -644,7 +644,9 @@ func (api *API) workspaceAgentCoordinate(rw http.ResponseWriter, r *http.Request
 			// This is a bug with unit tests that cancel the app context and
 			// cause this error log to be generated. We should fix the unit tests
 			// as this is a valid log.
-			if !xerrors.Is(err, context.Canceled) {
+			//
+			// The pq error occurs when the server is shutting down.
+			if !xerrors.Is(err, context.Canceled) && err.Error() != "pq: canceling statement due to user request" {
```

Comment:
Could we do something similar to this? https://github.com/coder/coder/blob/main/coderd/database/errors.go

I’m thinking, if we move away from pq we’re not relying on magic strings that could silently fail and contain it in the DB package.

Reply:
Agreed!